### ISSUE TITLE
How to run Socket.IO (using node and express) events on HTTP2 protocol?

### ISSUE THREAD

**Issue by masroorejaz** [nodejs/help#3875]
### Details

Hello Everyone,

I want to to transfer my Socket.IO code which is currently running on HTTPS 1.1 to HTTP2 and I am using node and express server. This tutorial (https://www.youtube.com/watch?v=b35Dcz91ItE&t=174s) has defined everything regarding setting up HTTP2 Socket really well. But it does not define how to use Socket.IO on and emit events in HTTP2. I have been searching for it since 2 days but no luck. Any help would be greatful. 

Thank you.

### Node.js version

17.8.2

### Operating system

Localhost win 10.
Linux on live server.

### Scope

Code and Runtime.

### Module and version

Not applicable.

**Comment by OliverEncoded** [comment]
Did you ever get this to work?
I'm having the same problem, I'm even fine with using a different protocol, but one that still has the same kind of event system.

Please respond, it's going on 3 days I've been struggling.

**Comment by masroorejaz** [comment]
Nope.

I am now running 2 Mongodb and React App on HTTP2 and Socket.IO on HTTPS 1.1 since even I have got no luck running Socket.IO in HTTP2

I have also posted it reddit as well. While it does not provide me with any solution it describe me in great details on why there is no use of running Socket.IO on HTTP2:
https://www.reddit.com/r/node/comments/uzt56e/comment/iacijg2/

If you still wants to run Socket.IO on HTTP2 then try ChatGPT as I didn't have ChatGPT when I was working on it. ChatGPT might actually able to help you out.
